lib/hooks/use-cart.ts

```diff
@@ -0,0 +1,30 @@
+import useBigCommerceCart from '@bigcommerce/storefront-data-hooks/cart/use-cart'
+import useBigCommercePrice from '@bigcommerce/storefront-data-hooks/use-price'
+
+export const useCart = () => {
+  const { data, error, isEmpty } = useBigCommerceCart()
+  const items = data?.line_items.physical_items ?? []
+
+  const { price: subtotal } = useBigCommercePrice(
+    data && {
+      amount: data.base_amount,
+      currencyCode: data.currency.code,
+    }
+  )
+  const { price: total } = useBigCommercePrice(
+    data && {
+      amount: data.cart_amount,
+      currencyCode: data.currency.code,
+    }
+  )
+
+  return {
+    items: data?.line_items.physical_items ?? [],
+    isLoading: data === undefined,
+    isError: error,
+    isEmpty: isEmpty && data === null,
+    subtotal,
+    total,
+    currency: data?.currency,
+  }
+}
```

Comment:
Move this to the page, we don't want to create a hook of the hook unless it's strictly required due to something that is app specific, is okay if it's more code.

Reply:
The reason I put this into a custom hook is that I think it opens the door to start thinking about multiple providers and alternate ways of fetching data, since the provider logic and the fetching mechanism would be encapsulated in here. 

Thoughts?

Reply:
@marbiano Each provider will have its own hooks ideally, like with BC, they should be used in the same way but the structure of the data is different, to make it work for multiple providers you would need to map the entire data into a completely different object because they don't follow a shared schema.

I think this is nice but not for now, again unless we really need it. The example should showcase how some external hooks for data fetching + UI components can create a powerful Commerce example. I don't want to start creating abstractions so soon. (We can always PR to the BC hooks if something needs to be done in a hook).

Reply:
@lfades makes sense, I'll put it back into the component.